static/js/lib/search.js

```diff
@@ -279,30 +279,76 @@ export const buildFacetSubQuery = (
   const facetClauses = []
   if (facets) {
     facets.forEach((values, key) => {
-      if (OBJECT_TYPE !== key && values && values.length > 0) {
-        facetClauses.push({
-          bool: {
-            // $FlowFixMe: shut up flow
-            should: values.map(value => ({
-              term: {
-                [key]: value
-              }
-            }))
+      const facetClausesForFacet = []
+
+      if (values && values.length > 0) {
+        addFacetClauseToArray(facetClauses, key, values)
+      }
+
+      // $FlowFixMe: we check for null facets earlier
+      facets.forEach((values2, key2) => {
```

Comment:
minor nitpick: maybe use something like `otherValues`, `otherKey` for the variable names